ui/src/app/utils/stores.tsx

```diff
@@ -46,20 +47,40 @@ interface CdrVersionStore {
 
 export const cdrVersionStore = atom<CdrVersionStore>({});
 
-export interface GenomicExtractionStore {
-  [workspaceNamespace: string]: GenomicExtractionJob[];
-}
-
-export const genomicExtractionStore = atom<GenomicExtractionStore>({});
-
-export const updateGenomicExtractionStore = (
+export const useGenomicExtractionJobs = (
   workspaceNamespace: string,
-  extractions: GenomicExtractionJob[]
-) => {
-  genomicExtractionStore.set({
-    ...genomicExtractionStore.get(),
-    [workspaceNamespace]: extractions,
-  });
+  workspaceId: string,
+  pollWhileNonTerminal = true
+) =>
+  useSWR(
+    `/api/workspaces/${workspaceNamespace}/${workspaceId}/genomicExtractionJobs`,
+    () =>
+      dataSetApi()
+        .getGenomicExtractionJobs(workspaceNamespace, workspaceId)
+        .then(({ jobs }) => jobs),
+    {
+      refreshInterval: (data) => {
```

Comment:
Refresh polling based on the current state of the data is pretty straightforward.